File: src/main/java/xyz/upperlevel/hgame/scenario/character/Character.java
package xyz.upperlevel.hgame.scenario.character;


public interface Character {
    /**
     * The name of the character.
     */
    String getName();

    /**
     * The {@code Character}'s texture path.
     */
    String getTexturePath();

    /**
     * Generates an instance of the {@link Character}'s {@link Actor}.
     * The {@link Actor} is the object that will populate the Scenario.
     */
    Actor personify(int id);
}

File: src/main/java/xyz/upperlevel/hgame/world/character/Character.kt
package xyz.upperlevel.hgame.world.character

import com.badlogic.gdx.physics.box2d.World


interface Character {
    /**
     * The name of the character.
     */
    val name: String

    /**
     * The `Character`'s texture path.
     */
    val texturePath: String

    /**
     * Generates an instance of the [Character]'s [Entity].
     * The [Entity] is the object that will populate the Scenario.
     */
    fun personify(id: Int, pworld: World): Entity
}
